mysql-test/suite/encryption/r/bug_ps_6106.result

```diff
@@ -0,0 +1,101 @@
+CREATE TABLE t1 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENGINE=InnoDB ENCRYPTION='KEYRING';
+CREATE TABLE t2 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENGINE=InnoDB COMPRESSION="zlib";
+CREATE TABLE t3 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENGINE=InnoDB;
+CREATE TABLE t4 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255)) ENGINE=InnoDB ENCRYPTION='N';
+CREATE TABLE t4_dup LIKE t4;
+SHOW CREATE TABLE t4_dup;
+Table	Create Table
+t4_dup	CREATE TABLE `t4_dup` (
+  `id` int(11) NOT NULL,
+  `a` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci
+CREATE DATABASE test_enc_n DEFAULT ENCRYPTION='N';
+USE test_enc_n;
+CREATE TABLE t5 (id INT NOT NULL PRIMARY KEY, a VARCHAR(255));
+USE test;
+CREATE PROCEDURE innodb_insert_proc(repeat_count INT)
+BEGIN
+DECLARE current_num INT;
+SET current_num = 0;
+WHILE current_num < repeat_count DO
+INSERT INTO t1 VALUES (current_num, REPEAT('foobar', 42));
+INSERT INTO t2 VALUES (current_num, REPEAT('temp'  , 42));
+INSERT INTO t3 VALUES (current_num, REPEAT('barfoo', 42));
+INSERT INTO t4 VALUES (current_num, REPEAT('secret', 42));
+INSERT INTO t4_dup VALUES (current_num, REPEAT('fbar', 42));
+INSERT INTO test_enc_n.t5 VALUES (current_num, REPEAT('secsec', 42));
+SET current_num = current_num + 1;
+END WHILE;
+END//
+COMMIT;
+SET autocommit = 0;
+CALL innodb_insert_proc(10000);
+COMMIT;
+SET autocommit = 1;
+# Wait max 2 min for key encryption threads to encrypt all spaces
+include/assert.inc [Make sure t4 is not encrypted]
+include/assert.inc [Make sure t4_dup is not encrypted]
+include/assert.inc [Make sure t5 is encrypted]
+ALTER TABLE t1 ENGINE InnoDB;
+SHOW CREATE TABLE t1;
+Table	Create Table
+t1	CREATE TABLE `t1` (
+  `id` int(11) NOT NULL,
+  `a` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci ENCRYPTION='KEYRING' ENCRYPTION_KEY_ID=0
+ALTER TABLE t2 ENGINE InnoDB;
+SHOW CREATE TABLE t2;
+Table	Create Table
+t2	CREATE TABLE `t2` (
+  `id` int(11) NOT NULL,
+  `a` varchar(255) DEFAULT NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci COMPRESSION='zlib' ENCRYPTION_KEY_ID=0
```

Comment:
ENCRYPTION=KEYRING is for tables created with explicit ENCRYPTION='KEYRING' clause. This is for foreground encryption like with MK : ENCRYPTION='Y'. Those tables cannot be decrypted by background threads. We see ENCRYPTION_KEY_ID - which shows us the id of the key encryption threads used to encrypt table.

Reply:
Ok, this looks like separate issue to discuss. My question is ideally we should hide ENCRYPTION_KEY_ID from SHOW CREATE TABLE.

There should be two approached IMHO.
1) After background threads encrypt table, ENCRYPTION='KEYRING' should be shown along with ENCRYPTION_KEY_ID. Whats the benefit in hiding them? 
or
2) Hide everything related to background keyring encryption. (Although I wonder whats the benefit)

Reply:
ENCRYPTION='KEYRING' is for tables that were either CREATEd or ALTERed with encryption clause = 'KEYRING'. Those tables are always encrypted with KEYRING and never decrypted by encryption threads. In other words they are always skipped by encryption threads.
ENCRYPTION_KEY_ID on the other hand is used by encryption threads - this is the key_id that is going to be used by encryption threads when encrypting tables - thus it can be shown in SHOW CREATE TABLE.
If we shown ENCRYPTION='KEYRING' for tables that were encrypted by encryption threads we would be saying that those tables where CREATEd or ALTERed to ENCRYPTION='KEYRING' which is not true and has different meaning.

Reply:
I will remove ENCRYPTION_KEY_ID if it was not set during CREATE or ATLER table.

Reply:
Okay. I see them as design issues inherited from upstream(MariaDB). There is too much detail (what is explicit and implicit, which tables are encrypted by background) left to user to possibly reduce scope.

You need not fix this in this PR

Reply:
I have already done it in this PR :)